pyro/contrib/funsor/handlers/plate_messenger.py

```diff
@@ -315,10 +315,10 @@ def _pyro_sample(self, msg):
             assert msg["name"].endswith(str(self._indices))
             msg["name"] = msg["name"][:-len(str(self._indices))] + str(self._suffix)
         if str(self._suffix) != str(self._suffixes[-1]):
-            # do not trace auxiliary vars
-            msg["infer"]["_do_not_trace"] = True
+            # _do_not_score: record these sites when tracing for use with replay,
+            # but do not include them in ELBO computation.
+            msg["infer"]["_do_not_score"] = True
             msg["infer"]["is_auxiliary"] = True
```

Comment:
`is_auxiliary` can be removed too I think. The only reason it is there is because of the assertion statement in
https://github.com/pyro-ppl/pyro/blob/b5e4a7b10a445b947bc5d670bcc9eb43480fec8c/pyro/poutine/trace_messenger.py#L124